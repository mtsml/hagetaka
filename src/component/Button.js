import React from 'react'
import { MDBBtn } from 'mdbreact'
import { THEME_COLOR } from '../util/const'

const Button = (props) => {
    return (
        <MDBBtn 
            disabled={props.disabled}
            className='mt-2 hbtn'
            color={props.color||THEME_COLOR}
            onClick={props.onClick}
        >
            {props.children}
        </MDBBtn>
    )
}

export default Button