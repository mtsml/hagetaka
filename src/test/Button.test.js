import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../component/Button';

describe('スナップショット: ', () => {
    it('正常', () => {
        const props = {
            disabled: false, 
            color: 'red', 
            onClick: function(){}
        };
        const tree = renderer.create(<Button {...props}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
})