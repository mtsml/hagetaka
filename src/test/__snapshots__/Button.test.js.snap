// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショット:  正常 1`] = `
<button
  className="btn-red btn Ripple-parent mt-2 h-btn"
  data-test="button"
  disabled={false}
  onClick={[Function]}
  onMouseUp={[Function]}
  onTouchStart={[Function]}
  type="button"
>
  <div
    className="Ripple "
    data-test="waves"
    style={
      Object {
        "height": "0px",
        "left": "0px",
        "top": "0px",
        "width": "0px",
      }
    }
  />
</button>
`;
